sql/analyzer/indexed_joins.go

```diff
@@ -645,13 +649,76 @@ func addHashJoins(m *memo.Memo) error {
 	})
 }
 
+func addSlidingRangeJoin(m *memo.Memo) error {
+	return memo.DfsRel(m.Root(), func(e memo.RelExpr) error {
+		switch e.(type) {
+		case *memo.InnerJoin, *memo.LeftJoin:
+		default:
+			return nil
+		}
+
+		join := e.(memo.JoinRel).JoinPrivate()
+		if len(join.Filter) != 1 {
+			return nil
+		}
+
+		filter := join.Filter[0]
+
+		switch f := filter.(type) {
+		case *memo.Between:
+			if !(satisfiesScalarRefs(f.Value.Scalar, join.Left) &&
+				satisfiesScalarRefs(f.Min.Scalar, join.Right) &&
+				satisfiesScalarRefs(f.Max.Scalar, join.Right)) {
+				return nil
+			}
+			// TODO: Is this safe? If the expression references multiple columns, does this reference one
```

Comment:
selection conditions:
- we have one rel with a LessThan/GreaterThan filter combo that sandwiches the same attribute on the other rel
- there are a variety of exclusion conditions for complex expressions that would interfere with range join (ex: `x + 1 <= max AND x > min + 7` is OK because expressions are monotonic increasing (`isWeaklyMonotonic`))
- there are a variety of child node exclusion conditions for each relation (I think tableAlias, resolvedTable, Filter are the 3 OK)
- we have a `sortedIndexForTableCol` for finding appropriate sorted indexes

The most novel steps are the first two, because we don't have any other joins that try to stitch together filters into a Between (which I think have been normalized in a prior analyzer step). Exclusion conditions can be empty to start, we just want to get TableAlias and ResolvedTable into the execution phase for perf testing.

Assembling, we have 1) min expr, 2) max expr, 3) target expr, 4) range table + index (IndexScan), 5) target table + index (IndexScan). Building will fix indexes for 1,2,3, and then (4) and (5) will both be converted into `IndexedTableAccess` for a range spanning NULL->infinity. For mergeJoin I was lazy and put all of the various filter expressions into the join condition in a specific order that I could predictably unpack on the other side.

Reply:
Handling sided-ness and left joins will be another set of edge cases

Reply:
This should be handled in the new PR now.